pika/adapters/select_connection.py

```diff
@@ -189,24 +162,27 @@ class SelectPoller(object):
     to override the update_handler and start methods for additional types.
 
     """
-    TIMEOUT = 1
+    TIMEOUT = 5
+    # if the poller used MS specify 1000
+    TIMEOUT_MULT = 1
 
-    def __init__(self, fileno, handler, events, state_manager):
+    def __init__(self, state_manager):
         """Create an instance of the SelectPoller
 
-        :param int fileno: The file descriptor to check events for
-        :param method handler: What is called when an event happens
-        :param int events: The events to look for
         :param method state_manager: The method to manage state
 
         """
-        self.fileno = fileno
-        self.events = events
-        self.open = True
-        self._handler = handler
-        self._timeouts = []
+        self.filenos = dict()
+        self.r_fds = set()
+        self.w_fds = set()
+        self.e_fds = set()
+        self.stopping = False
+        self._timeouts = {}
+        self._next_timeout = None
+        r_interupt, self.w_interupt = os.pipe()
```

Comment:
@wjps, there needs to be some way for `r_interupt` and `self.w_interupt` to be closed. Otherwise, and app that creates/destroys many SelectConnection instances over its lifetime will end up leaking lots of kernel resources. The pollers should probably have a `close()` method that should be called reliably; perhaps from `SelectConnection._adapter_disconnect()`:

```
 def _adapter_disconnect(self):
        """Disconnect from the RabbitMQ broker"""
        try:
            if self.socket:
                self.ioloop.remove_handler(self.socket.fileno())
            super(SelectConnection, self)._adapter_disconnect()
        finally:
            if self.ioloop:
                self.ioloop.close() # and it's implementation calls the underlying poller's `close()` method
```